### PR TITLE
Added sanity checks to :meth:`~.Mobject.add_to_back` for Mobjects

### LINKED ISSUE #1099: Mobjects add_to_back is missing sanity checks
## Description of bug / unexpected behavior
the Mobject.add method contains sanity checks to make sure, no submobject gets added twice and self is not in self.submobjects.
Mobjext.add_to_back is missing this functionality.

## Expected behavior
Adding self should produce a ValueError: Mobject cannot contain self.
Addin a Mobject multiple times should result in it beeing just once in self.submobjects

## How to reproduce the issue
<details><summary>Code for reproducing the problem</summary>

```py
a = Mobject()
b = Line()
a.add_to_back(a)
a.add_to_back(b,b)
print(a.submobjects)
# results in [Mobject, Line, Line]
```

</details>

### PR DESCRIPTION
**Pull request by RickyC0626**
## Changelog / Overview

Add Mobject `add_to_back` sanity checks:
- Raises ValueError when Mobject tries to add itself
- Raises TypeError when a non-Mobject is added
- Filters out incoming duplicate submobjects if at least one instance of that submobject exists in the list

Closes #1099 

## Motivation

Additional checks were needed for `add_to_back` when dealing with `Mobject`, similar to `add`

## Explanation for Changes

See changelog / overview above

## Testing Status

Running `poetry` on **WSL: Ubuntu-20.04**
```py
def test_vmob_add_to_back():
    """Test the Mobject add_to_back method."""
    a = VMobject()
    b = Line()
    c = "text"
    with pytest.raises(ValueError):
        # Mobject cannot contain self
        a.add_to_back(a)
    with pytest.raises(TypeError):
        # All submobjects must be of type Mobject
        a.add_to_back(c)

    # No submobject gets added twice
    a.add_to_back(b)
    a.add_to_back(b, b)
    assert len(a.submobjects) == 1
```

## Further Comments

Duplication prevention may need additional testing for edge cases such as:
```py
a.add_to_back(b, b, b)
a.add_to_back(b, b)
```

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [x] My new functions/classes either have a docstring or are private
- [x] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [x] The PR title is descriptive enough
- [x] The PR is labeled correctly
- [x] The changelog entry is completed if necessary
- [x] Newly added functions/classes either have a docstring or are private
- [x] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

**Review by huguesdevimeux — APPROVED** [review]
SGTM, thanks for this !

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -10,4 +10,8 @@ repos:
   - repo: https://gitlab.com/PyCQA/flake8
     rev: 3.9.0
     hooks:
-    -   id: flake8
+      - id: flake8
+  - repo: https://github.com/PyCQA/isort
+    rev: 5.8.0
+    hooks:
+      - id: isort
```
Comment (kolibril13) [inline .pre-commit-config.yaml:17]:
```suggestion
```

**Review by behackl — APPROVED** [review]
Thank you for your contribution!
